pydantic/config.py

```diff
@@ -592,10 +592,90 @@ class Transaction(BaseModel):
     The format of JSON serialized timedeltas. Accepts the string values of `'iso8601'` and
     `'float'`. Defaults to `'iso8601'`.
 
-    - `'iso8601'` will serialize timedeltas to ISO 8601 durations.
+    !!! warning
+        In v2.11+ it is recommended to use the [`ser_json_datetime`][pydantic.config.ConfigDict.ser_json_datetime]
+        setting instead of `ser_json_timedelta`. This setting will be deprecated in v3.
+
+    - `'iso8601'` will serialize timedeltas to [ISO 8601 durations](https://en.wikipedia.org/wiki/ISO_8601#Durations).
     - `'float'` will serialize timedeltas to the total number of seconds.
     """
 
+    val_datetime_unit: Literal['seconds', 'milliseconds', 'infer']
+    """
+    The unit to assume for validating numeric input for datetime-like types. This includes:
+
+    - [`datetime.datetime`][]
+    - [`datetime.date`][]
+    - [`datetime.time`][]
+    - [`datetime.timedelta`][]
+
+    Defaults to `'infer'`.
+    The "epoch" reference below refers to the Unix epoch, which is 1970-01-01 00:00:00 UTC.
+
+    - `'seconds'` will validate date or time numeric inputs as seconds since the epoch.
+    - `'milliseconds'` will validate date or time numeric inputs as milliseconds since the epoch.
+    - `'infer'` will infer the unit from the string numeric input on unix time:
+        i.e. seconds (if >= -2e10 and <= 2e10) or milliseconds (if < -2e10or > 2e10) since the epoch.
+    """
+
+    ser_datetime_unit: Literal['seconds', 'milliseconds', 'infer']
+    """
+    The unit to use for serializing datetime-like types. This includes:
+
+    - [`datetime.datetime`][]
+    - [`datetime.date`][]
+    - [`datetime.time`][]
+    - [`datetime.timedelta`][]
+
+    This only applies when serializing to `'float'`. ISO8601 durations use a predefined format.
+    Defaults to `'infer'`.
+    The "epoch" reference below refers to the Unix epoch, which is 1970-01-01 00:00:00 UTC.
+
+    - `'seconds'` will validate date or time numeric inputs as seconds since the epoch.
+    - `'milliseconds'` will validate date or time numeric inputs as milliseconds since the epoch.
+    - `'infer'` will infer the unit from the string numeric input on unix time:
+        i.e. seconds (if >= -2e10 and <= 2e10) or milliseconds (if < -2e10or > 2e10) since the epoch.
```

Comment:
```suggestion
        i.e. seconds (if >= -2^10 and <= 2^10) or milliseconds (if < -2^10 or > 2^10) since the epoch.
```

This will be rendered properly in docs, as per https://squidfunk.github.io/mkdocs-material/reference/formatting/#sub-and-superscripts.